parcels/fieldset.py

```diff
@@ -40,6 +40,11 @@ def from_data(cls, data, dimensions, transpose=True, mesh='spherical',
 
         :param data: Dictionary mapping field names to numpy arrays.
                Note that at least a 'U' and 'V' numpy array need to be given
+               1. If data shape is [xdim, ydim], [xdim, ydim, zdim], [xdim, ydim, tdim] or [xdim, ydim, zdim, tdim],
+                  Use the flag transpose=True (default value)
+               2. If data shape is [ydim, xdim], [zdim, ydim, xdim], [tdim, ydim, xdim] or [tdim, zdim, ydim, xdim],
+                  Use the flat transpose=False
+               3. If data has any other shape, you first need to reorder it
```

Comment:
add `to dimensions [ydim, xdim], [zdim, ydim, xdim], [tdim, ydim, xdim] or [tdim, zdim, ydim, xdim], whichever is relevant for your data set`

Same of course for docstring in `field.py`